lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        if let Some(timestamp) = parsed.timestamp {
+            let naive = NaiveDateTime::from_timestamp(
+                f64::trunc(timestamp) as i64,
+                f64::fract(timestamp) as u32,
+            );
+            log.insert(
+                log_schema().timestamp_key(),
+                DateTime::<Utc>::from_utc(naive, Utc),
+            );
+        } else {
+            log.insert(log_schema().timestamp_key(), Utc::now());
+        }
+
+        if let Some(level) = parsed.level {
+            log.insert(LEVEL, level);
+        }
+        if let Some(facility) = &parsed.facility {
+            log.insert(FACILITY, facility.to_string());
+        }
+        if let Some(line) = &parsed.line {
+            log.insert(LINE, *line);
+        }
+        if let Some(file) = &parsed.file {
+            log.insert(FILE, file.to_string());
+        }
+
+        if let Some(add) = &parsed.additional_fields {
+            for (key, val) in add.iter() {
+                // per GELF spec, filter out _id
+                if key == "_id" {
+                    continue;
+                }
+                // per GELF spec, Additional field names must be characters dashes or dots
+
+                // drop fields names that don't match the GELF spec
+                if !self.regex.is_match(key) {
+                    continue;
+                }
+
+                // per GELF spec, values to additional fields can be strings or numbers
+                if val.is_string() {
+                    log.insert(path!(key.as_str()), val.as_str());
+                } else if val.is_u64() {
+                    log.insert(path!(key.as_str()), val.as_u64());
+                } else if val.is_i64() {
+                    log.insert(path!(key.as_str()), val.as_i64());
+                } else if val.is_f64() {
+                    match ordered_float::NotNan::new(val.as_f64().unwrap()) {
+                        Ok(float) => {
+                            log.insert(path!(key.as_str()), float);
+                        }
+                        Err(e) => return Err(e.to_string().into()),
+                    }
+                }
+            }
+        }
+
+        Ok(event)
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct GelfMessage {
+    version: String,
+    host: String,
+    short_message: String,
+    full_message: Option<String>,
+    timestamp: Option<f64>,
+    level: Option<u8>,
+    facility: Option<String>,
+    line: Option<usize>,
+    file: Option<String>,
+    #[serde(flatten)]
+    additional_fields: Option<HashMap<String, serde_json::Value>>,
+}
+
+impl Deserializer for GelfDeserializer {
+    fn parse(&self, bytes: Bytes) -> vector_core::Result<SmallVec<[Event; 1]>> {
+        let line = std::str::from_utf8(&bytes)?;
+        let line = line.trim();
+
+        let parsed: GelfMessage = serde_json::from_str(line)?;
+        let event = self.insert_fields_from_gelf(&parsed)?;
```

Comment:
Nit: the method name here, `insert_fields_from_gelf` seems a bit weird to me now since I'd expect it to be taking a mutable reference and modifying it. Here we are taking a reference and returning a new struct. Maybe `self.message_to_event`?